src/Datadog.Trace/TraceContext.cs

```diff
@@ -104,9 +105,10 @@ public void CloseSpan(Span span)
 
             lock (_spans)
             {
+                _spans.Add(span);
                 _openSpans--;
 
-                if (_openSpans == 0)
+                if (_openSpans == 0 || ShouldTriggerPartialFlush())
```

Comment:
@kevingosse just to make sure I understand, by setting the "top level" metric on the correct span, the agent will accept these partial traces and there's nothing else we need to do? And the behavior exists on the 0.4 endpoint? Also, it's worth documenting on what version of the Agent this functionality was enabled, perhaps in the ConfigurationKey's

Reply:
That's correct.

And yes, I will add an agent version check before merging

Reply:
Actually, if that's OK with you, I'm going to merge the PR as-is and add the agent check in a separate PR.
The agent check requires a version of the agent that hasn't been released yet, so I'd like to publish a pre-release before implementing it.